index.js

```diff
@@ -12,17 +13,81 @@ const xpiZipFilename = stringToBytes('META-INF/mozilla.rsa');
 const oxmlContentTypes = stringToBytes('[Content_Types].xml');
 const oxmlRels = stringToBytes('_rels/.rels');
 
-const fileType = input => {
+/**
+ * Determine file type from Node.js stream
+ * @param {Readable} stream
+ * @returns {Promise<{ext, mime}|undefined>}
+ * @public
+ */
+async function fromStream(stream) {
+	const tokenizer = await strtok3.fromStream(stream);
+	try {
+		return await fromTokenizer(tokenizer);
+	} finally {
+		await tokenizer.close();
+	}
+}
+
+/**
+ * Determine file type from Buffer or Uint8Array.
+ * @param {Buffer | Uint8Array | ArrayBuffer} input - Partial file.
+ * @returns {Promise<{ext, mime}|undefined>} Promise with file-type
+ * @public
+ */
+function fromBuffer(input) {
 	if (!(input instanceof Uint8Array || input instanceof ArrayBuffer || Buffer.isBuffer(input))) {
 		throw new TypeError(`Expected the \`input\` argument to be of type \`Uint8Array\` or \`Buffer\` or \`ArrayBuffer\`, got \`${typeof input}\``);
 	}
 
-	const buffer = input instanceof Uint8Array ? input : new Uint8Array(input);
+	const buffer = input instanceof Buffer ? input : Buffer.from(input);
```

Comment:
Note https://github.com/sindresorhus/file-type/commit/dc611b4c1293b7df9f9416d234e9b8935494519d

Reply:
The reason I am using `Buffer` is that I need pass a [Buffer](https://nodejs.org/api/buffer.html) to `strtok3.fromBuffer`.

It took me a while to figure out why UInt8Array instead of [Buffer](https://nodejs.org/api/buffer.html), please correct me if I am wrong, [browser compatibility](https://github.com/sindresorhus/file-type/pull/227#issuecomment-517741187) was the reason to use [Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array).

To me it looks like the Buffer dependency in browser side code becomes less of a problem as module bundlers getting better in shimming this common dependency.

I am using the `buffer.copy` in [strtok3](https://github.com/Borewit/strtok3). Also [token-types](https://github.com/Borewit/token-types) is using Buffer specific functions, such as `readUInt16LE`, `readUInt16BE`, `readInt8` etc.

So [buffer](https://nodejs.org/api/buffer.html) functionality is actually being use. I could get rid of the Buffer dependency, in [strtok3](https://github.com/Borewit/strtok3) & [token-types](https://github.com/Borewit/token-types), but it means the functionality has to be duplicated using alternative ways. I am not sure if that is the way to go. As we scan files a bit deeper to determine the right file-type, I expect buffer's method will more and more used for decoding.

Reply:
It's not a problem for me personally. It's just some browser users complained about size. But we can deal with it later if someone complains.

Reply:
I am sure few users won't like it.